src/components/spatial/SpatialSubscriber.js

```diff
@@ -70,7 +74,13 @@ export default function SpatialSubscriber(props) {
     setCellHighlight,
   }] = useCoordination(COMPONENT_COORDINATION_TYPES.spatial, coordinationScopes);
 
-  const [autoLayers, setAutoLayers] = useState({});
+  const [autoLayers, setAutoLayers] = useState({
+    [dataset]: [
+      loaders[dataset].loaders.cells?.url ? DEFAULT_CELLS_LAYER : null,
+      loaders[dataset].loaders.molecules?.url ? DEFAULT_MOLECULES_LAYER : null,
+      loaders[dataset].loaders.neighborhoods?.url ? DEFAULT_NEIGHBORHOODS_LAYER : null,
+    ].filter(i => i),
```

Comment:
These filters are needed so we don't have `null` values - wasn't sure there was a better way not go about this

Reply:
```suggestion
    ].filter(Boolean),
```

I think you can do this too